### DISCUSSION TITLE
Python Usage Guidance

### DISCUSSION THREAD

**Discussion by dssangan** [zinggAI/zingg#440] [Q&A] [answered]
Hello,

I'm fairly new to Entity Resolution analytics and came across Zingg. I am working within databricks and utilized this documentation: https://docs.zingg.ai/zingg/working-with-python. I have the library installed, but getting following error when trying part of this code: https://github.com/zinggAI/zingg/blob/main/examples/febrl/FebrlExample.py

![image](https://user-images.githubusercontent.com/32006591/192612041-7ffd488f-8b0c-40ea-929a-fd13cab51426.png)

Am I missing any steps in proper usage of python library?

Thanks!

**Comment by sonalgoyal** [comment]
Hi @dssangan, thanks for trying Zingg! Can you try downloading the release and adding the jar to to the cluster?

**Reply by dssangan** [reply]
I added the zingg snapshot 0.3.3 jar file to cluster. Currently running into following error, could you explain the difference between zingg library available through PyPi and the purpose of the jar?

![image](https://user-images.githubusercontent.com/32006591/192848615-3824cc26-d34c-4688-993e-82fd49c98923.png)

I looked online for the error I received and found that it is due to environment variables not being set appropriately. And currently I don't know how to do that within Databricks. Attaching couple screenshot of my databricks cluster and libraries that I'm adding externally on top of pre-existing that comes along with cluster.

![image](https://user-images.githubusercontent.com/32006591/192857118-2b5046ec-4f68-4ad5-a5e7-0b8697fe7789.png)
![image](https://user-images.githubusercontent.com/32006591/192857153-99b198c7-5b55-4e5b-80dd-c428ac77d8a9.png)

Let me know if you have any further guidance/suggestions that I could try out.

**Comment by sonalgoyal** [comment, marked as answer]
Can you please try with 0.3.4 version? The pypi has the python wrapper, but the jar is still needed as that’s what the python api invokes. The python api has been released with 0.3.4

**Reply by dssangan** [reply]
Thank you @sonalgoyal !! I am able to use the library now!!
